grid_map_core/src/Polygon.cpp

```diff
@@ -83,4 +87,93 @@ void Polygon::resetTimestamp()
   timestamp_ = 0.0;
 }
 
+Polygon Polygon::convexHullCircles(const Position center1, const Position center2, const double radius)
+{
+  Eigen::Vector2d centerToVertex, centerToVertexTemp;
+  centerToVertex = center2 - center1;
+  centerToVertex.normalize();
+  centerToVertex *= radius;
+  const int nVertices = 10;
+
+  grid_map::Polygon polygon;
+  for (int j=0; j<nVertices; j++) {
+    double theta = M_PI_2 + j*M_PI/(nVertices-1);
+    Eigen::Rotation2D<double> rot2d(theta);
+    centerToVertexTemp = rot2d.toRotationMatrix()*centerToVertex;
+    polygon.addVertex(center1+centerToVertexTemp);
+  }
+  for (int j=0; j<nVertices; j++) {
+    double theta = 3*M_PI_2 + j*M_PI/(nVertices-1);
+    Eigen::Rotation2D<double> rot2d(theta);
+    centerToVertexTemp = rot2d.toRotationMatrix()*centerToVertex;
+    polygon.addVertex(center2+centerToVertexTemp);
+  }
+  return polygon;
+}
+
+Polygon Polygon::convexHullCircle(const Position center, const double radius)
+{
+  Eigen::Vector2d centerToVertex(radius, 0.0), centerToVertexTemp;
+  const int nVertices = 20;
+
+  grid_map::Polygon polygon;
+  for (int j=0; j<nVertices; j++) {
+    double theta = j*2*M_PI/(nVertices-1);
+    Eigen::Rotation2D<double> rot2d(theta);
+    centerToVertexTemp = rot2d.toRotationMatrix()*centerToVertex;
+    polygon.addVertex(center+centerToVertexTemp);
+  }
+  return polygon;
+}
+
+Polygon Polygon::convexHull(Polygon& polygon1, Polygon& polygon2)
+{
+  std::vector<Eigen::Vector2d> vertices1 = polygon1.getVertices();
+  std::vector<Eigen::Vector2d> vertices2 = polygon2.getVertices();
+  std::vector<Eigen::Vector2d> vertices = vertices1;
+  for (const auto& vertice : vertices2) {
```

Comment:
This could be done nicer (less copying) with http://stackoverflow.com/questions/3177241/best-way-to-concatenate-two-vectors